utils/ChromiumDownloader.js

```diff
@@ -217,3 +224,17 @@ function downloadFile(url, destinationPath, progressCallback) {
 function extractZip(zipPath, folderPath) {
   return new Promise(fulfill => extract(zipPath, {dir: folderPath}, fulfill));
 }
+
+function applyAgent(opt) {
+  const proxy_url = getProxyForUrl(opt.href);
+
+  if (proxy_url)
+  {
+    const parsedProxy = URL.parse(proxy_url);
+
+    if (parsedProxy.protocol === 'http:')
+      parsedProxy.secureProxy = false;
+
+    opt.agent = new ProxyAgent(parsedProxy);
```

Comment:
I don't like that we pass result of URL.parse over to the ProxyAgent constructor. I had to check what's the output of `URL.parse` and make sure it doesn't clash with some options of `ProxyAgent`.

Let's make a method `requestOptions` which returns options for the `https.request` module, and will handle proxy inside. Something along the lines: 

```js
function requestOptions(url, method = 'GET') {
  const parsedURL = URL.parse(url);
  const result = {
    method,
    host: parsedURL.host,
    path: parsedURL.pathname
  };
  const proxyURL = getProxyForUrl(url);
  if (proxyURL) {
    const parsedProxyURL = URL.parse(proxyURL);
    result.agent = new ProxyAgent({
      host: parsedProxyURL.host,
      port: parsedProxyURL.port,
      secureProxy: parsedProxyURL.protocol === 'https:'
    });
  }
  return result;
}
```